docs/12-system-tests.md

```diff
@@ -0,0 +1,164 @@
+ID:
+
+```bash
+./scripts/test/tearDown.sh <instance-id>
+```
+
+This will remove your deployment, including any ECR repos, S3 buckets and local files that have been generated.
+
+## Acceptance tests
+
+We use an acceptance test suite to judge releasability of the system. These tell us whether the system can perform its
+functions in a deployed environment, and help us understand whether changes to the code have increased or decreased the
+performance. The tests sit in the Maven module `system-test/system-test-suite`. There are scripts to run these tests
+under `scripts/test/maven`. We run this test suite nightly.
+
+These tests run in JUnit, and use the class `SleeperSystemTest` as the entry point. That class defines the domain
+specific language (DSL) for working with a deployed instance of Sleeper in JUnit. Please review the comment at the top
+of that class, and look at the tests in that module for examples.
+
+This test suite contains both feature tests, and performance tests which work with a larger bulk of data. By default,
+the suite skips the performance tests. You can run that like this:
+
+```bash
+./scripts/test/maven/buildDeployTest.sh <short-id> <vpc> <subnets>
+```
+
+The short ID will be used to generate the instance IDs of Sleeper instances deployed for the tests. The feature tests
+will run on one Sleeper instance, but the performance tests will deploy more. A separate CDK stack will also be deployed
+with `SystemTestStandaloneApp`, for resources shared between tests. This will use the short ID as its stack name.
+
+After the tests, the instances will remain deployed. If you run again with the same short ID, the same instances will
+be used without redeploying them. You can tear down all resources associated with a short ID like this:
+
+```bash
+./scripts/test/maven/tearDown.sh <short-id> <instance-ids>
+```
+
+This requires you to know which instance IDs are deployed. These will be in the form `<short-id>-<test-id>`. You can
+find the different test IDs in the SystemTestInstance class. These are the identifiers associated with each enum value,
+and you can find references to those enum values to find the tests that use them. When a test runs, it deploys an
+instance with the associated instance ID if one does not exist.
+
+### Performance tests
+
+You can run specific performance tests like this:
+
+```bash
+./scripts/build/buildForTest.sh # This is not necessary if you used buildDeployTest.sh or have already built the system
+./scripts/test/maven/performanceTest.sh <short-id> <vpc> <subnets> CompactionPerformanceIT,IngestPerformanceIT
+```
+
+Performance tests use an ECS cluster for generating data, and will deploy that with `SystemTestStandaloneApp`. For
+non-performance tests, that stack will still be used, but the data generation cluster will be disabled. When an instance
+of that has been deployed for a short ID, the test suite will reuse it without checking whether it has the data
+generation cluster enabled.
```

Comment:
I'm finding it a bit hard to follow what stack is being referenced from this paragraph. It may be worth replacing the ambiguous bits like `and will deploy that with`, `that stack will still be used` and `When and instance of that has been deployed` with more specific references.

Reply:
Thanks, I'll change it.